src/components/application_manager/src/commands/button_notification_to_mobile.cc

```diff
@@ -0,0 +1,209 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "application_manager/commands/button_notification_to_mobile.h"
+namespace application_manager {
+
+namespace commands {
+namespace app_mngr = application_manager;
+
+SDL_CREATE_LOG_VARIABLE("ButtonNotificationToMobile")
+
+ButtonNotificationToMobile::ButtonNotificationToMobile(
+    const app_mngr::commands::MessageSharedPtr& message,
+    app_mngr::ApplicationManager& application_manager,
+    app_mngr::rpc_service::RPCService& rpc_service,
+    app_mngr::HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handler)
+    : CommandNotificationImpl(message,
+                              application_manager,
+                              rpc_service,
+                              hmi_capabilities,
+                              policy_handler) {}
+
+ButtonNotificationToMobile::~ButtonNotificationToMobile() {}
+
+bool ButtonNotificationToMobile::DoesAppIDExist() const {
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager::strings;
+  return (*message_)[msg_params].keyExists(strings::app_id);
+}
+
+void ButtonNotificationToMobile::HandleCustomButton(
+    app_mngr::ApplicationSharedPtr app) {
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager::strings;
+  // app_id is mandatory for CUSTOM_BUTTON notification
+  if (!DoesAppIDExist()) {
+    SDL_LOG_ERROR("CUSTOM_BUTTON mobile notification without app_id.");
+    return;
+  }
+
+  // custom_button_id is mandatory for CUSTOM_BUTTON notification
+  if (!(*message_)[msg_params].keyExists(hmi_response::custom_button_id)) {
+    SDL_LOG_ERROR(
+        "CUSTOM_BUTTON mobile notification without custom_button_id.");
+    return;
+  }
+
+  if (!app) {
+    SDL_LOG_ERROR("Application doesn't exist.");
+    return;
+  }
+
+  const uint32_t custom_btn_id =
+      (*message_)[msg_params][hmi_response::custom_button_id].asUInt();
+
+  if (!app->IsSubscribedToSoftButton(custom_btn_id)) {
+    SDL_LOG_ERROR("Application doesn't subscribed to this custom_button_id.");
+    return;
+  }
+
+  if (mobile_api::HMILevel::HMI_NONE ==
+      app->hmi_level(app->GetSoftButtonWindowID(custom_btn_id))) {
+    SDL_LOG_WARN("CUSTOM_BUTTON mobile notification is allowed only "
+                 << "in FULL, LIMITED or BACKGROUND hmi level");
+    return;
+  }
+
+  SendButtonNotification(app);
+  return;
+}
+
+void ButtonNotificationToMobile::HandleOKButton(
+    app_mngr::ApplicationSharedPtr app) {
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager;
+  SDL_LOG_DEBUG("OK button received");
+
+  const auto subscribed_apps = SubscribedApps();
+  auto app_ptr = subscribed_apps.end();
```

Comment:
This variable does not have any sense here
It's better to declare `const auto` at line 111 and `const auto` at line 124